### PR TITLE
doc: Adds example and registry docs for resource_policy

### PR DESCRIPTION
**Pull request by EspenAlbert**
## Description

Adds example and registry docs for resource_policy

Link to any related issue(s): CLOUDP-272617

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.
- [x] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

**Review by marcosuma — APPROVED** [review]

**Comment by marcosuma** [comment]
non-blocking: would be nice to share this PR with eng and PM team for visibility

**Review by lantoli — APPROVED** [review]

File: docs/data-sources/resource_policies.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
+# Data Source: mongodbatlas_resource_policies
+
+`mongodbatlas_resource_policies` returns all resource policies in an organization.
+
+-> **NOTE**: Resource Policies are currently in Public Preview. You will need to
```
Comment (jwilliams-mongo) [inline docs/data-sources/resource_policies.md]:
```suggestion
-> **NOTE**: Resource Policies are currently in Public Preview. To use this feature, you must take the following actions:
```

File: docs/data-sources/resource_policy.md [hunk 2]
```diff
@@ -0,0 +1,134 @@
+# Data Source: mongodbatlas_resource_policy
+
+`mongodbatlas_resource_policy` describes a resource policy in an organization.
+
+-> **NOTE**: Resource Policies are currently in Public Preview. You will need to
```
Comment (jwilliams-mongo) [inline docs/data-sources/resource_policy.md]: same change as above

File: docs/data-sources/resource_policies.md [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+### Read-Only
+
+- `resource_policies` (Attributes List) (see [below for nested schema](#nestedatt--resource_policies))
+
+<a id="nestedatt--resource_policies"></a>
+### Nested Schema for `resource_policies`
+
+Read-Only:
+
+- `created_by_user` (Attributes) The user that last updated the atlas resource policy. (see [below for nested schema](#nestedatt--resource_policies--created_by_user))
```
Comment (jwilliams-mongo) [inline docs/data-sources/resource_policies.md]:
```suggestion
- `created_by_user` (Attributes) The user that last updated the Atlas resource policy. (see [below for nested schema](#nestedatt--resource_policies--created_by_user))
```

File: docs/data-sources/resource_policies.md [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+- `created_date` (String) Date and time in UTC when the atlas resource policy was created.
```
Comment (jwilliams-mongo) [inline docs/data-sources/resource_policies.md]:
```suggestion
- `created_date` (String) Date and time in UTC when the Atlas resource policy was created.
```

File: docs/data-sources/resource_policies.md [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+- `id` (String) Unique 24-hexadecimal digit string that identifies an atlas resource policy.
```
Comment (jwilliams-mongo) [inline docs/data-sources/resource_policies.md]:
```suggestion
- `id` (String) Unique 24-hexadecimal digit string that identifies an Atlas resource policy.
```

File: docs/data-sources/resource_policies.md [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+- `last_updated_by_user` (Attributes) The user that last updated the atlas resource policy. (see [below for nested schema](#nestedatt--resource_policies--last_updated_by_user))
```
Comment (jwilliams-mongo) [inline docs/data-sources/resource_policies.md]:
```suggestion
- `last_updated_by_user` (Attributes) The user that last updated the Atlas resource policy. (see [below for nested schema](#nestedatt--resource_policies--last_updated_by_user))
```

File: docs/data-sources/resource_policies.md [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+- `last_updated_date` (String) Date and time in UTC when the atlas resource policy was last updated.
+- `name` (String) Human-readable label that describes the atlas resource policy.
+- `org_id` (String) Unique 24-hexadecimal digit string that identifies the organization that contains your projects. Use the [/orgs](#tag/Organizations/operation/listOrganizations) endpoint to retrieve all organizations to which the authenticated user has access.
+- `policies` (Attributes List) List of policies that make up the atlas resource policy. (see [below for nested schema](#nestedatt--resource_policies--policies))
+- `version` (String) A string that identifies the version of the atlas resource policy.
```
Comment (jwilliams-mongo) [inline docs/data-sources/resource_policies.md]:
```suggestion
- `last_updated_date` (String) Date and time in UTC when the Atlas resource policy was last updated.
- `name` (String) Human-readable label that describes the Atlas resource policy.
- `org_id` (String) Unique 24-hexadecimal digit string that identifies the organization that contains your projects. Use the [/orgs](#tag/Organizations/operation/listOrganizations) endpoint to retrieve all organizations to which the authenticated user has access.
- `policies` (Attributes List) List of policies that make up the Atlas resource policy. (see [below for nested schema](#nestedatt--resource_policies--policies))
- `version` (String) A string that identifies the version of the Atlas resource policy.
```

File: docs/data-sources/resource_policy.md [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+  value = { for policy in data.mongodbatlas_resource_policies.this.resource_policies : policy.name => policy.id }
+}
+```
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- `id` (String) Unique 24-hexadecimal digit string that identifies an atlas resource policy.
```
Comment (jwilliams-mongo) [inline docs/data-sources/resource_policy.md]:
```suggestion
- `id` (String) Unique 24-hexadecimal digit string that identifies an Atlas resource policy.
```

File: docs/data-sources/resource_policy.md [hunk 9]
```diff
@@ -0,0 +1,134 @@
...
+
+### Read-Only
+
+- `created_by_user` (Attributes) The user that last updated the atlas resource policy. (see [below for nested schema](#nestedatt--created_by_user))
+- `created_date` (String) Date and time in UTC when the atlas resource policy was created.
+- `last_updated_by_user` (Attributes) The user that last updated the atlas resource policy. (see [below for nested schema](#nestedatt--last_updated_by_user))
+- `last_updated_date` (String) Date and time in UTC when the atlas resource policy was last updated.
+- `name` (String) Human-readable label that describes the atlas resource policy.
+- `policies` (Attributes List) List of policies that make up the atlas resource policy. (see [below for nested schema](#nestedatt--policies))
+- `version` (String) A string that identifies the version of the atlas resource policy.
```
Comment (jwilliams-mongo) [inline docs/data-sources/resource_policy.md]:
```suggestion
- `created_by_user` (Attributes) The user that last updated the Atlas resource policy. (see [below for nested schema](#nestedatt--created_by_user))
- `created_date` (String) Date and time in UTC when the Atlas resource policy was created.
- `last_updated_by_user` (Attributes) The user that last updated the Atlas resource policy. (see [below for nested schema](#nestedatt--last_updated_by_user))
- `last_updated_date` (String) Date and time in UTC when the Atlas resource policy was last updated.
- `name` (String) Human-readable label that describes the Atlas resource policy.
- `policies` (Attributes List) List of policies that make up the Atlas resource policy. (see [below for nested schema](#nestedatt--policies))
- `version` (String) A string that identifies the version of the Atlas resource policy.
```

File: docs/resources/resource_policy.md [hunk 10]
```diff
@@ -0,0 +1,145 @@
+# Resource: mongodbatlas_resource_policy
+
+`mongodbatlas_resource_policy` provides a Resource Policy resource. The resource lets you create, edit and delete resource policies to prevent misconfigurations and reduce the need for corrective interventions in your organization.
+
+-> **NOTE**: Resource Policies are currently in Public Preview. You will need to
```
Comment (jwilliams-mongo) [inline docs/resources/resource_policy.md]: same comment as in first file

File: docs/resources/resource_policy.md [hunk 11]
```diff
@@ -0,0 +1,145 @@
...
+```
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- `name` (String) Human-readable label that describes the atlas resource policy.
+- `org_id` (String) Unique 24-hexadecimal digit string that identifies the organization that contains your projects. Use the [/orgs](#tag/Organizations/operation/listOrganizations) endpoint to retrieve all organizations to which the authenticated user has access.
+- `policies` (Attributes List) List of policies that make up the atlas resource policy. (see [below for nested schema](#nestedatt--policies))
```
Comment (jwilliams-mongo) [inline docs/resources/resource_policy.md]:
```suggestion
- `name` (String) Human-readable label that describes the Atlas resource policy.
- `org_id` (String) Unique 24-hexadecimal digit string that identifies the organization that contains your projects. Use the [/orgs](#tag/Organizations/operation/listOrganizations) endpoint to retrieve all organizations to which the authenticated user has access.
- `policies` (Attributes List) List of policies that make up the Atlas resource policy. (see [below for nested schema](#nestedatt--policies))
```

File: docs/resources/resource_policy.md [hunk 12]
```diff
@@ -0,0 +1,145 @@
...
+
+### Read-Only
+
+- `created_by_user` (Attributes) The user that last updated the atlas resource policy. (see [below for nested schema](#nestedatt--created_by_user))
+- `created_date` (String) Date and time in UTC when the atlas resource policy was created.
+- `id` (String) Unique 24-hexadecimal digit string that identifies an atlas resource policy.
+- `last_updated_by_user` (Attributes) The user that last updated the atlas resource policy. (see [below for nested schema](#nestedatt--last_updated_by_user))
+- `last_updated_date` (String) Date and time in UTC when the atlas resource policy was last updated.
+- `version` (String) A string that identifies the version of the atlas resource policy.
```
Comment (jwilliams-mongo) [inline docs/resources/resource_policy.md]:
```suggestion
- `created_by_user` (Attributes) The user that last updated the Atlas resource policy. (see [below for nested schema](#nestedatt--created_by_user))
- `created_date` (String) Date and time in UTC when the Atlas resource policy was created.
- `id` (String) Unique 24-hexadecimal digit string that identifies an Atlas resource policy.
- `last_updated_by_user` (Attributes) The user that last updated the Atlas resource policy. (see [below for nested schema](#nestedatt--last_updated_by_user))
- `last_updated_date` (String) Date and time in UTC when the Atlas resource policy was last updated.
- `version` (String) A string that identifies the version of the Atlas resource policy.
```

File: examples/mongodbatlas_resource_policy/README.md [hunk 13]
```diff
@@ -0,0 +1,12 @@
+# MongoDB Atlas Provider -- Atlas Resource Policy
+This example creates three different resource policies in an organization.
+
+**NOTE**: Resource Policies are currently in Public Preview. You will need to
```
Comment (jwilliams-mongo) [inline examples/mongodbatlas_resource_policy/README.md]: same comment as on first file

File: templates/data-sources/resource_policies.md.tmpl [hunk 14]
```diff
@@ -0,0 +1,13 @@
+# {{.Type}}: {{.Name}}
+
+`{{.Name}}` returns all resource policies in an organization.
+
+-> **NOTE**: Resource Policies are currently in Public Preview. You will need to
```
Comment (jwilliams-mongo) [inline templates/data-sources/resource_policies.md.tmpl]: same comment as on first file

File: templates/data-sources/resource_policy.md.tmpl [hunk 15]
```diff
@@ -0,0 +1,13 @@
+# {{.Type}}: {{.Name}}
+
+`{{.Name}}` describes a resource policy in an organization.
+
+-> **NOTE**: Resource Policies are currently in Public Preview. You will need to
```
Comment (jwilliams-mongo) [inline templates/data-sources/resource_policy.md.tmpl]: same comment as on first file

File: templates/resources/resource_policy.md.tmpl [hunk 16]
```diff
@@ -0,0 +1,21 @@
+# {{.Type}}: {{.Name}}
+
+`{{.Name}}` provides a Resource Policy resource. The resource lets you create, edit and delete resource policies to prevent misconfigurations and reduce the need for corrective interventions in your organization.
+
+-> **NOTE**: Resource Policies are currently in Public Preview. You will need to
```
Comment (jwilliams-mongo) [inline templates/resources/resource_policy.md.tmpl]: same comment as on first file

File: docs/data-sources/resource_policies.md (hunk 1 above)
Comment (EspenAlbert) [reply docs/data-sources/resource_policies.md]: Changed in a8851acdb8c44933b67c874a3160f5b6938a5ba2

File: docs/data-sources/resource_policies.md (hunk 3 above)
Comment (EspenAlbert) [reply docs/data-sources/resource_policies.md]: Changed in a8851acdb8c44933b67c874a3160f5b6938a5ba2

File: docs/data-sources/resource_policies.md (hunk 5 above)
Comment (EspenAlbert) [reply docs/data-sources/resource_policies.md]: Changed in a8851acdb8c44933b67c874a3160f5b6938a5ba2

File: docs/data-sources/resource_policies.md (hunk 6 above)
Comment (EspenAlbert) [reply docs/data-sources/resource_policies.md]: Changed in a8851acdb8c44933b67c874a3160f5b6938a5ba2

File: docs/data-sources/resource_policies.md (hunk 7 above)
Comment (EspenAlbert) [reply docs/data-sources/resource_policies.md]: Changed in a8851acdb8c44933b67c874a3160f5b6938a5ba2

File: docs/data-sources/resource_policy.md (hunk 9 above)
Comment (EspenAlbert) [reply docs/data-sources/resource_policy.md]: Changed in a8851acdb8c44933b67c874a3160f5b6938a5ba2

File: docs/resources/resource_policy.md (hunk 11 above)
Comment (EspenAlbert) [reply docs/resources/resource_policy.md]: Changed in a8851acdb8c44933b67c874a3160f5b6938a5ba2

File: docs/resources/resource_policy.md (hunk 12 above)
Comment (EspenAlbert) [reply docs/resources/resource_policy.md]: Changed in a8851acdb8c44933b67c874a3160f5b6938a5ba2

File: templates/data-sources/resource_policy.md.tmpl (hunk 15 above)
Comment (EspenAlbert) [reply templates/data-sources/resource_policy.md.tmpl]: Changed in a8851acdb8c44933b67c874a3160f5b6938a5ba2

File: templates/resources/resource_policy.md.tmpl (hunk 16 above)
Comment (EspenAlbert) [reply templates/resources/resource_policy.md.tmpl]: Changed in a8851acdb8c44933b67c874a3160f5b6938a5ba2

File: templates/data-sources/resource_policies.md.tmpl (hunk 14 above)
Comment (EspenAlbert) [reply templates/data-sources/resource_policies.md.tmpl]: Changed in a8851acdb8c44933b67c874a3160f5b6938a5ba2

File: examples/mongodbatlas_resource_policy/README.md (hunk 13 above)
Comment (EspenAlbert) [reply examples/mongodbatlas_resource_policy/README.md]: Changed in a8851acdb8c44933b67c874a3160f5b6938a5ba2

File: docs/data-sources/resource_policy.md (hunk 2 above)
Comment (EspenAlbert) [reply docs/data-sources/resource_policy.md]: Changed in a8851acdb8c44933b67c874a3160f5b6938a5ba2

File: docs/data-sources/resource_policies.md (hunk 4 above)
Comment (EspenAlbert) [reply docs/data-sources/resource_policies.md]: Changed in a8851acdb8c44933b67c874a3160f5b6938a5ba2

**Review by jwilliams-mongo — APPROVED** [review]

**Pull request merged**